cmd/process-agent/collector_test.go

```diff
@@ -345,3 +351,214 @@ func TestIgnoreResponseBody(t *testing.T) {
 		})
 	}
 }
+
+func TestCollectorRunCheckWithRealTime(t *testing.T) {
+	cfg := config.NewDefaultAgentConfig()
+	check := &mocks.CheckWithRealTime{}
+
+	c, err := NewCollector(cfg, []checks.Check{})
+	assert.NoError(t, err)
+
+	results := api.NewWeightedQueue(1, 1024)
+	rtResults := api.NewWeightedQueue(1, 1024)
+
+	standardOption := checks.RunOptions{
+		RunStandard: true,
+	}
+
+	result := &checks.RunResult{
+		Standard: []model.MessageBody{
+			&model.CollectorProc{},
+		},
+	}
+
+	check.On("RunWithOptions", mock.Anything, mock.Anything, standardOption).Once().Return(result, nil)
+	check.On("Name").Return("foo")
+	check.On("RealTimeName").Return("bar")
+
+	c.runCheckWithRealTime(
+		check,
+		results,
+		rtResults,
+		standardOption,
+	)
+
+	assert.Equal(t, results.Len(), 1)
+	item, ok := results.Poll()
+	assert.True(t, ok)
+	assert.Equal(t, item.Type(), "foo")
+
+	assert.Equal(t, rtResults.Len(), 0)
+
+	rtResult := &checks.RunResult{
+		RealTime: []model.MessageBody{
+			&model.CollectorProc{},
+		},
+	}
+
+	rtOption := checks.RunOptions{
+		RunStandard: true,
```

Comment:
What's the difference between `standardOption` and `rtOption`? They look the same to me, but you're returning different results from the mock.

Reply:
Hmmm, how does this test actually assert 😬 let me investigate. I meant RunRealTime here.

Reply:
Ah ok, so the option values will be ignored since we are only passing them to the mock.